### PR TITLE
fix(platform): do not open menu on disabled menu button and split menu button

### PR DESCRIPTION
**Pull request by DeepakSap14**
#### Please provide a link to the associated issue.
Closes SAP/fundamental-ngx#5513

#### Please provide a brief summary of this pull request.
Even when menu button/split menu button is disabled, clicking on it opens the associated menu.

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [x] Run npm run build-pack-library and test in external application

Documentation checklist:
- [na] update `README.md`
- [na] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

**Review by sKudum — APPROVED** [review]

**Comment by maxym-valor** [comment]
Approved!

**Review by Betrozov — APPROVED** [review]

**Review by platon-rov — APPROVED** [review]

**Review by N1XUS — APPROVED** [review]

File: libs/platform/src/lib/components/menu-button/menu-button.component.html [hunk 1]
```diff
@@ -6,10 +6,9 @@
     [label]="label"
     [compact]="contentDensity === 'compact'"
     [glyph]="icon"
-    [disabled]="disabled"
```
Comment (dimamarksman) [inline libs/platform/src/lib/components/menu-button/menu-button.component.html:9]: Why did you remove `[disabled]` button state?
You are trying to move this `is-disabled` class to the host element itself, but I believe it's not safe.
`.fd-button.is-disabled` has some css overrides but in this case you miss it.
Let's avoid it and try to find another way.

**Review by dimamarksman — CHANGES_REQUESTED** [review]

File: libs/platform/src/lib/components/menu-button/menu-button.component.ts [hunk 2]
```diff
@@ -30,10 +39,26 @@ export class MenuButtonComponent extends BaseComponent {
...
+    tabindex = 0;
+
     constructor(_cd: ChangeDetectorRef) {
         super(_cd);
     }
 
+    /** @hidden */
+    ngOnInit(): void {
+        this.menuButtonDisabled = this.disabled;
+        this.tabindex = this.disabled ? -1 : 0;
```
Comment (dimamarksman) [inline libs/platform/src/lib/components/menu-button/menu-button.component.ts:59]: probably we need a getter for it to keep it up to date cause `disabled` state might happen after `OnInit` phase

File: libs/platform/src/lib/components/menu-button/menu-button.component.html (hunk 1 above)
Comment (dimamarksman) [reply libs/platform/src/lib/components/menu-button/menu-button.component.html:9]: Also in this case you are introducing A11 issue as well.

Comment (DeepakSap14) [reply libs/platform/src/lib/components/menu-button/menu-button.component.html:9]: keeping [disabled] now.
needed to disabled cursor/pointer-event at menu-button level, to avoid click when disabled. corrected it now.

File: libs/platform/src/lib/components/menu-button/menu-button.component.ts (hunk 2 above)
Comment (DeepakSap14) [reply libs/platform/src/lib/components/menu-button/menu-button.component.ts:70]: using getter and setter now.

Comment (dimamarksman) [reply libs/platform/src/lib/components/menu-button/menu-button.component.ts:70]: It's not what I meant.
In your case when `disabled` gets changed after OnInit `tabindex` and `menuButtonDisabled` remains the same and doesn't reflect this change. Please correct me if I'm wrong.

What I mean:
```typescript
@HostBinding('class.menu-button-disabled')
get  menuButtonDisabled(): boolean {
    return this.disabled;
}

get tabindex(): number {
    return this.disabled ? -1 : 0;
}
```

Comment (DeepakSap14) [reply libs/platform/src/lib/components/menu-button/menu-button.component.ts]: okay.. understood your point. have corrected the code now.

**Review by dimamarksman — APPROVED** [review]
Thanks

**Comment by maxym-valor** [comment]
Approved!

**Review by InnaAtanasova — DISMISSED** [review]
`menu-button-disabled` this is not a valid class name. It should be coming from fund-styles and have the `fd-` prefix.

**Comment by DeepakSap14** [comment]
> `menu-button-disabled` this is not a valid class name. It should be coming from fund-styles and have the `fd-` prefix.

@InnaAtanasova 
this class needed in platform only. should i create a component in styles for 1 class? what do you think?
corrected class name.

**Comment by droshev** [comment]
Closing old PRs - feel free to reopen it if/when needed
